apps/showcase/src/components/showcase/forms/forms-pres.component.ts

```diff
@@ -0,0 +1,138 @@
+
+  constructor() {
+    this.submitTrigger$ = this.submitForm$.asObservable();
+    this.translations = translations;
+    this.personalInfo = { name: '', dateOfBirth: this.formatDate(Date.now()) };
+    this.personalInfoFormControl = new UntypedFormControl(this.personalInfo);
+    this.emergencyContact = { name: '', phone: '', email: '' };
+    this.emergencyContactFormControl = new UntypedFormControl(this.emergencyContact);
+    this.personalInfoValidators = {
+      global: formsPresValidatorGlobal(this.forbiddenName, translations.globalForbiddenName, translations.globalForbiddenNameLong, { name: this.forbiddenName }),
+      fields: {
+        dateOfBirth: dateCustomValidator(translations.dateInThePast)
+      }
+    };
+    this.emergencyContactValidators = {
+      global: formsPresValidatorGlobal(this.forbiddenName, translations.globalForbiddenName, translations.globalForbiddenNameLong, { name: this.forbiddenName })
+    };
+  }
+
+  private formatDate(dateTime: number) {
+    return formatDate(dateTime, 'yyyy-MM-dd', 'en-GB');
+  }
+
+  public ngOnInit() {
+    this.subscription.add(this.submitTrigger$.subscribe((_value) => this.submitAction()));
+  }
+
+  public ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  public submitForm() {
+    this.submitForm$.next(true);
+  }
+
+  /** This will store the function to make the personal info form as dirty and touched */
+  public _markPersonalInfoInteraction: () => void = () => {};
+  /** This will store the function to make the emergency contact form as dirty and touched */
+  public _markEmergencyContactInteraction: () => void = () => {};
+
+  /**
+   * Register the function to be called to mark the personal info form as touched and dirty
+   *
+   * @param fn
+   */
+  public registerPersonalInfoInteraction(fn: () => void) {
+    this._markPersonalInfoInteraction = fn;
+  }
+
+  /**
+   * Register the function to be called to mark the personal emergency contact form as touched and dirty
+   *
+   * @param fn
+   */
+  public registerEmergencyContactInteraction(fn: () => void) {
+    this._markEmergencyContactInteraction = fn;
+  }
+
+  /** submit function */
+  public submitAction() {
+    if (this.submitTrigger$ && this.firstSubmit) {
+      this._markPersonalInfoInteraction();
+      this._markEmergencyContactInteraction();
+      this.firstSubmit = false;
+    }
+    const isValid = !this.personalInfoFormControl.errors && !this.emergencyContactFormControl.errors;
+    if (isValid) {
+      // eslint-disable-next-line no-console
+      console.log('FORMS COMPONENT: personal info form status', this.personalInfoFormControl.status);
+      // eslint-disable-next-line no-console
+      console.log('FORMS COMPONENT: emergency contact form status', this.emergencyContactFormControl.status);
+      // eslint-disable-next-line no-console
+      console.log('FORMS COMPONENT: submit logic here', this.personalInfoFormControl.value, this.emergencyContactFormControl.value);
+    }
```

Comment:
Could you create an issue to track the `errors` case with the example of the store usage ?

Reply:
Would you like me to update the PR and add the store to track the errors? Or would you prefer for me to create an issue?

Reply:
another issue would be cleaner imo.